src/main/scala/com.snowplowanalytics.iglu/client/Resolver.scala

```diff
@@ -234,9 +235,14 @@ case class Resolver(
           cache.store(schemaKey, collectErrors(schemaKey, errors, tried).fail)
         case repo :: repos => {
           repo.lookupSchema(schemaKey) match {
-            case Success(Some(schema)) => cache.store(schemaKey, schema.success)
-            case Success(None)         => recurse(schemaKey, errors, tried.::(repo), repos)
-            case Failure(e)            => recurse(schemaKey, errors.::(e), tried.::(repo), repos)
+            case Success(Some(schema)) if isValid(schema) =>
+              cache.store(schemaKey, schema.success)
+            case Success(Some(schema)) =>
+              recurse(schemaKey, errors ++ getErrors(schema), tried.::(repo), repos)
+            case Success(None) =>
+              recurse(schemaKey, errors, tried.::(repo), repos)
+            case Failure(e) =>
+              recurse(schemaKey, errors.::(e), tried.::(repo), repos)
```

Comment:
I don't know why this file was using `listTail.::(listHead)` instead of `listHead :: listTail`. We are changing the indentation so we could change that too...